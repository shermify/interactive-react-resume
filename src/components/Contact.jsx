import React from 'react';
import PropTypes from 'prop-types';
import Section from './Section';

const Contact = (props) => {
  const { lead, quoteBy, text, phone, email } = props;
  return (
    <Section id="contact" title="Contact" bgcolor="#f2f2f2" lead={lead} quoteBy={quoteBy}>
      <div>
        <p>{text}</p>
        <p>
          Tel: {phone}<br />
          Email: {email.split('@')[0]}<span style={{ display: 'none' }}>f3hy78Y*&97hy</span>@{email.split('@')[1]}<br />
          <a href="https://linkedin.com/in/dlsherman">Linkedin</a>
        </p>
      </div>
    </Section>
  );
};

Contact.propTypes = {
  lead: PropTypes.string.isRequired,
  quoteBy: PropTypes.string,
  text: PropTypes.string.isRequired,
  phone: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
};

Contact.defaultProps = {
  quoteBy: '',
};


export default Contact;
