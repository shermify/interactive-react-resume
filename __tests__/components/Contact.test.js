import React from 'react';
import renderer from 'react-test-renderer';
import Contact from '../../src/components/Contact';

describe('<Contact />', () => {
  const testData = {
    lead: 'test lead',
    quoteBy: 'test quote',
    text: 'jkvsdhf kjh kjfhkjfjk j jhk',
    phone: '773-551-3085',
    email: 'example@example.com',
  };

  it('renders correctly', () => {
    const tree = renderer.create(
      <Contact {...testData} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
