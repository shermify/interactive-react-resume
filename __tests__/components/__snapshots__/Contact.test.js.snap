// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Contact /> renders correctly 1`] = `
<section
  id="contact"
  style={
    Object {
      "backgroundColor": "#f2f2f2",
      "color": "#333",
    }
  }
>
  <div
    className="container section"
  >
    <h2>
      Contact
    </h2>
    <p
      className="lead"
    >
      test lead
      <br />
      test quote
    </p>
    <hr />
    <div>
      <div>
        <p>
          jkvsdhf kjh kjfhkjfjk j jhk
        </p>
        <p>
          Tel: 
          773-551-3085
          <br />
          Email: 
          example
          <span
            style={
              Object {
                "display": "none",
              }
            }
          >
            f3hy78Y*&97hy
          </span>
          @
          example.com
          <br />
          <a
            href="https://linkedin.com/in/dlsherman"
          >
            Linkedin
          </a>
        </p>
      </div>
    </div>
  </div>
</section>
`;
